20928 Add details about HTMLMetaElement API usage with media (#21511)
mdn/content: files/en-us/web/api/htmlmetaelement/media/index.md

---
title: HTMLMetaElement.media
slug: Web/API/HTMLMetaElement/media
page-type: web-api-instance-property
tags:
  - API
  - HTML DOM
  - HTMLMetaElement
  - Property
  - Reference
browser-compat: api.HTMLMetaElement.media
---

{{APIRef("HTML DOM")}}

> The **`HTMLMetaElement.media`** property allows you to specify the media for which the metadata applies.
> This is useful for the `theme-color` property as it allows you to set the color of the browser's toolbar or UI elements depending on the media the page is displayed on.

Reviewer: ```suggestion
The **`HTMLMetaElement.media`** property enables specifying the media for `theme-color` metadata. 

The `theme-color` metatag enables setting the color of the browser's toolbar or UI in supporting browsers and operating systems. The `media` property enables setting different theme colors for different `media` values.
```

## Value

A string.

## Examples

### Setting the theme color

```js
var meta = document.createElement("meta");
meta.name = "theme-color";
meta.content = "#3c790a";
document.head.appendChild(meta);

> ```

Reviewer: this example doesn't use media.

Reviewer: Most meta tags can only be 'used' once. The color scheme can be used multiple times. Maybe create and example where two meta tags are created, one with a color for small devices and the next for all others. I believe the order of meta tag acceptance is first one that is a match wins and the rest are ignored, but would double check.

Author: added in f6cc7cbb18f062adddf89f49045474ad16e9f7f3

### Setting the theme color for dark mode

When a user has specified a dark mode in their operating system, the `media` property can be used to set a different `theme-color`:

```js
var meta = document.createElement("meta");
meta.name = "theme-color";
meta.content = "#3c790a";
meta.media = "(prefers-color-scheme: dark)";
document.head.appendChild(meta);
```

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

> - {{HTMLElement("meta")}}

Reviewer: How about a link to an article on possible `media` values?

Author: Added in 80d84fc081

---
title: HTMLMetaElement.media
slug: Web/API/HTMLMetaElement/media
page-type: web-api-instance-property
tags:
  - API
  - HTML DOM
  - HTMLMetaElement
  - Property
  - Reference
browser-compat: api.HTMLMetaElement.media
---

{{APIRef("HTML DOM")}}

The **`HTMLMetaElement.media`** property enables specifying the media for `theme-color` metadata.

> The `theme-color` property enables setting the color of the browser's toolbar or UI in supporting browsers and operating systems.

Reviewer 2: ```suggestion
The `theme-color` property enables setting the color of the browser's toolbar or UI in browsers and operating systems that support this property.
```
or maybe "in supporting browsers and operating systems" can be dropped because it is implied?

The `media` property enables setting different theme colors for different `media` values.

## Value

A string.

## Examples

### Setting the theme color for dark mode

When a user has specified a dark mode in their operating system, the `media` property can be used to set a different `theme-color`:

```js
var meta = document.createElement("meta");
meta.name = "theme-color";
meta.content = "#3c790a";
meta.media = "(prefers-color-scheme: dark)";
document.head.appendChild(meta);
```

### Setting theme colors by device size

> Most meta properties can only be used once, however `theme-color` can be used multiple times if there are unique `media` values provided.

Reviewer 2: ```suggestion
Most meta properties can be used only once. However, `theme-color` can be used multiple times if unique `media` values are provided.
```

> This example adds two meta elements, one with a `theme-color` for all devices and another for small screens.
> The order of matching the `media` query matters, so the more specific query should be added later in the document:

Reviewer 2: ```suggestion
This example adds two meta elements with a `theme-color`, one for all devices and another for small screens.
The order of matching the `media` query matters, so the more specific query should be added later in the document, as shown below:
```

```js
// Add a theme-color for all devices
meta = document.createElement("meta");
meta.name = "theme-color";
meta.content = "#ffffff";
document.head.appendChild(meta);

> // Add a theme-color specific to small devices

Reviewer 2: ```suggestion
// Add a theme-color for small devices
```
